### PR TITLE
Initial sphinx setup

### PR DESCRIPTION
**Pull request by csdev**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# general stuff
+*.swp
+*.swo
+*.DS_Store
+.directory
+
```
Comment (c-nichols) [inline .gitignore]: .pyc et al -- https://github.com/github/gitignore/blob/master/Python.gitignore

File: docs/Makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
+#
```
Comment (myfashionhub) [inline docs/Makefile]: Empty line

Comment (csdev) [inline docs/Makefile]: this is an auto-generated file

File: docs/Makefile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# Put it first so that "make" without argument is like "make help".
+help:
+	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+.PHONY: help Makefile
+
+# Catch-all target: route all unknown targets to Sphinx using the new
+# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
+%: Makefile
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```
Comment (apontejose) [inline docs/Makefile]: missing blank line

File: .gitignore (hunk 1 above)
Comment (csdev) [inline .gitignore]: currently, there's no build step here that's outputting python object files

Comment (c-nichols) [inline .gitignore]: ok -- assumed since BUILDSPHINX is python and conf.py is .py, it would output a compiled file...

**Pull request merged**